packages/gestalt-codemods/generic-codemods/utils.js

```diff
@@ -5,261 +5,278 @@ import {
-  importDeclaration.specifiers.map((node) =>
-    node.imported.name === previousComponentName
-      ? j.importSpecifier(j.identifier(nextComponentName))
-      : node,
-  );
-
-type SortImportedNamesType = {| importSpecifiers: Array<ImportSpecifierType> |};
+  jSXCollection,
+  prop,
+  value,
+}: {
+  j: JSCodeShift,
+  jSXCollection: Collection,
+  prop: string,
+  value?: string,
+}): Collection => {
+  if (typeof value === 'string') {
+    return jSXCollection.find(j.JSXAttribute, { name: { name: prop }, value: { value } });
+  }
 
-/**
- * sortImportedNames: Returns a sorted list of named imports
- * E.g. input: import { Pog, Box } from 'gestalt' >> output: import { Box, Pog } from 'gestalt'
- */
-const sortImportedNames = ({
-  importSpecifiers,
-}: SortImportedNamesType): Array<ImportSpecifierType> =>
-  importSpecifiers.sort((a, b) => a.imported.name.localeCompare(b.imported.name));
+  if (typeof value === 'number') {
+    return jSXCollection.find(j.JSXAttribute, {
+      value: {
+        type: 'JSXExpressionContainer',
+        expression: {
+          type: 'Literal',
+          value,
+        },
+      },
+    });
+  }
 
-type ReplaceImportNodePathType = {|
-  j: JSCodeShift,
-  nodePath: NodePathType,
-  importSpecifiers: Array<ImportSpecifierType>,
-  importPath: string,
-|};
+  if (typeof value === 'boolean') {
+    return value
+      ? jSXCollection.find(j.JSXAttribute, {
+          value: null,
+        })
+      : jSXCollection.find(j.JSXAttribute, {
+          value: {
+            type: 'JSXExpressionContainer',
+            expression: {
+              type: 'Literal',
+              value,
+            },
+          },
+        });
+  }
 
-/**
- * replaceImportNodePath: Replaces an import declaration node with an updated one
- * E.g. previousComponentName = Box & nextComponentName =  RenamedBox
- * input: <Box /> >> output: <RenamedBox />
- */
+  if (!value) return jSXCollection.find(j.JSXAttribute, { name: { name: prop } });
 
-const replaceImportNodePath = ({ nodePath, importSpecifiers }: ReplaceImportNodePathType): void => {
-  // TODO: find alternative to prevent reassignment
-  // eslint-disable-next-line no-param-reassign
-  nodePath.node.specifiers = importSpecifiers;
+  return jSXCollection;
 };
 
-type RenameJSXElementType = {| JSXNode: JSXNodeType, nextComponentName: string |};
-
 /**
- * renameJSXElement: Renames the JSX element with the name value provided
- * E.g. previousComponentName = Box & nextComponentName =  RenamedBox
- * input: <Box /> >> output: <RenamedBox />
+ * deepCloneNode: Returns a collection containing the Gestalt import declaration node-path
  */
-const renameJSXElement = ({ JSXNode, nextComponentName }: RenameJSXElementType): void => {
-  // TODO: implement deep cloning or prevent reassignment
-  const newJSXNode = { ...JSXNode };
-  newJSXNode.openingElement.name.name = nextComponentName;
+const deepCloneNode = <T>({ node }: { node: T }): T => JSON.parse(JSON.stringify(node));
```

Comment:
Nodes are by definition serializable, right? Will this ever hit something like a function or `undefined` that would cause it to bonk?

Reply:
you can't deep clone node paths (cos there are circular issues). I think the AST nodes are fine. I can try to stress test it. I'll merge first and stress test it to find edge cases.